libgnucash/engine/qof-string-cache.cpp

```diff
@@ -43,64 +45,42 @@ static QofLogModule log_module = QOF_MOD_UTIL;
 /* =================================================================== */
 /* The QOF string cache                                                */
 /*                                                                     */
-/* The cache is a GHashTable where a copy of the string is the key,    */
-/* and a ref count is the value                                        */
+/* The cache is an unordered_map where a copy of the string is the     */
+/* key, and a ref count is the value                                   */
 /* =================================================================== */
 
-static GHashTable* qof_string_cache = NULL;
-
-static GHashTable*
-qof_get_string_cache(void)
-{
-    if (!qof_string_cache)
-    {
-        qof_string_cache = g_hash_table_new_full(
-                               g_str_hash,               /* hash_func          */
-                               g_str_equal,              /* key_equal_func     */
-                               g_free,                   /* key_destroy_func   */
-                               g_free);                  /* value_destroy_func */
-    }
-    return qof_string_cache;
-}
+static std::unordered_map<std::string, guint> qof_string_cache;
 
 void
 qof_string_cache_init(void)
 {
-    (void)qof_get_string_cache();
+    qof_string_cache.clear();
 }
```

Comment:
Perhaps it's worth checking if this function is still necessary:

- As far as I can tell it's called only once exactly
- If I remember correctly an empty unordered map is the default state for a newly created one so clearing it doesn't do anything.

Reply:
Is there an (artificial) scenario where something could call init/destroy/init/destroy within 1 session? If so, both init and destroy should clear the map... :-/

Reply:
Not that I know of. Still in such imaginary scenario it would also suffice to have only one function, with a more sensible name like for example `qof_string_cache_clear`. Your init and destroy function currently do exactly the same and are an artefact of how things had to be dealt with in plain C.